### ISSUE TITLE
registerWidget requires preview component

### ISSUE THREAD

**Issue by chadfawcett** [netlify/netlify-cms#337]
**- Do you want to request a *feature* or report a *bug*?**
* bug

**- What is the current behavior?**
* Error http://facebook.github.io/react/docs/error-decoder.html?invariant=130&args[]=undefined&args[]=
```
reactProdInvariant.js:31 Uncaught (in promise) Error: Minified React error #130; visit http://facebook.github.io/react/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
```
![screen shot 2017-03-31 at 4 35 44 pm](https://cloud.githubusercontent.com/assets/3250463/24572719/273b2c4c-1630-11e7-9d71-79512a548531.png)

**- If the current behavior is a bug, please provide the steps to reproduce.**
* Follow the `registerWidget` tutorial in the [docs](https://www.netlifycms.org/docs/extending/#registerwidget)

**- What is the expected behavior?**
* As per the docs, the preview component is optional and should not cause an error.

**- Please mention your node.js, and operating system version.**
* Node: v6.10.1
* macOS Sierra

**Comment by erquhart** [comment]
That url in the error will take you to the complete error message, which is:
```
Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined.
```

If you paste the source of your widget I can help more, but it seems you're not passing a string as the first argument to the `h` function (which is actually [`React.createElement`](https://facebook.github.io/react/docs/react-api.html#createelement)).

**Comment by chadfawcett** [comment]
The issue only happens when I don't pass a component as a preview. Given the following script:

```javascript
var CategoriesControl = createClass({
  handleChange: function(e) {
    this.props.onChange(e.target.value.split(',').map((e) => e.trim()));
  },

  render: function() {
    var value = this.props.value;
    return h('input', { type: 'text', value: value ? value.join(', ') : '', onChange: this.handleChange });
  }
});

var CategoriesPreview = createClass({
  render: function() {
    return h('ul', null, this.props.value.map(function (category) {
        return h('li', null, category)
      })
    )
  }
})
```

This causes the error:
```javascript
CMS.registerWidget('categories', CategoriesControl);
```

While this does not cause the error:
```javascript
CMS.registerWidget('categories', CategoriesControl, CategoriesPreview);
```

As mentioned, the original error can be reproduced by using the source found in the docs.

**Comment by erquhart** [comment]
I believe you! But I can't reproduce. Just pasted in your exact code to be certain I wasn't missing anything, works a dream. Can you get a minimal reproduction into a public repo?

**Comment by chadfawcett** [comment]
I've put together a sample. If you follow the steps in the [README](https://github.com/chadfawcett/netlify-widget-issue-demo#readme), you should be able to replicate the issue.

https://github.com/chadfawcett/netlify-widget-issue-demo

**Comment by josephearl** [comment]
@chadfawcett I definitely see the error when trying out your example, I'll try to replicate locally with a dev build.

**Comment by erquhart** [comment]
I believe this was fixed through the kitchen sink example addition released in 0.4.3 - @chadfawcett please re-open if not.
